server/app/rest_api/rest.py

```diff
@@ -598,18 +592,15 @@ def post(self):
         if "varFilter" in args:
             set1_filter[Axis.VAR] = args["varFilter"]["filter"][Axis.VAR]
             set2_filter[Axis.VAR] = args["varFilter"]["filter"][Axis.VAR]
-        df1 = current_app.data.filter_dataframe(set1_filter, include_uns=False)
-        # TODO inverse
-        df2 = current_app.data.filter_dataframe(set2_filter, include_uns=False)
-        # exceeds size limit
-        if df1.shape[0] + df2.shape[0] > current_app.data.features["diffexp"]["interactiveLimit"]:
-            return make_response("Non-interactive request", HTTPStatus.FORBIDDEN)
         # mode
         count = args.get("count", None)
         try:
-            diffexp = current_app.data.diffexp(df1, df2, count)
-        except ValueError as ve:
-            return make_response(ve.message, HTTPStatus.BAD_REQUEST)
+            diffexp = current_app.data.diffexp(set1_filter, set2_filter, count,
+                                               current_app.data.features["diffexp"]["interactiveLimit"])
+        except ValueError as e:
```

Comment:
missing FilterError catch?

Reply:
good catch

Reply:
You can catch two exceptions at once like:

```
except (ValueError, FilterError) as ex:
    return make_response(ex.message, HTTPStatus.BAD_REQUEST)
```